### PR TITLE
chore(deps): update turbo and husky deps and configuration

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Update `turbo` and `husky` deps, remove deprecated API usage

#### Issue #, if available
NA

#### Description of how you validated changes
- `yarn && yarn build`
- verify commit blockage to `main`
- verify prevention of commit on lint error

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
   "private": true,
+  "packageManager": "yarn@1.22.22",
+  "name": "@aws-amplify/ui-monorepo",
```
Comment (calebpollman) [inline package.json:4]: Required by `turbo` upgrade

File: package.json [hunk 2]
```diff
@@ -30,7 +32,7 @@
     "vue-example": "yarn workspace @aws-amplify/ui-vue-example",
     "e2e": "yarn workspace @aws-amplify/ui-e2e",
     "environments": "yarn workspace @aws-amplify/ui-environments",
-    "prepare": "husky install",
+    "prepare": "husky",
```
Comment (calebpollman) [inline package.json:35]: Fixes:
```
$ husky install
install command is DEPRECATED
```

File: turbo.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "$schema": "https://turborepo.org/schema.json",
-  "baseBranch": "origin/main",
```
Comment (calebpollman) [inline turbo.json:3]: Fixes:
```
turbo_json_parse_error

  × failed to parse turbo json

Error:   × Found an unknown key `baseBranch`.
   ╭─[turbo.json:2:1]
 2 │   "$schema": "https://turborepo.org/schema.json",
 3 │   "baseBranch": "origin/main",
   ·   ────────────
 4 │   "pipeline": {
   ╰────
```

File: turbo.json [hunk 4]
```diff
@@ -1,7 +1,6 @@
...
-  "pipeline": {
+  "tasks": {
```
Comment (calebpollman) [inline turbo.json:3]: Required change for `turbo` upgrade

**Review by timngyn — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**